src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -192,8 +192,8 @@
 (defn initialize-schema
   "Ensure the database is migrated to the latest version, and warn if
   it's deprecated, log and exit if it's unsupported. We do this in a
-  single connection because HSQLDB seems to get confused if the
-  database doesn't exist but we open and close a connection without
+  single connection because HSQLDB seemed to get confused if the
```

Comment:
@senior I'd left it so that we'd know why the code was structured the way it is (if/when we come back along to rework things - which, given the System/exit, I'm hoping we do).  I could also just move that text from the doc string to a comment - or were you suggesting I should go ahead and rework the code now?